src/controllers/colormapperjsproxy.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <QScriptContext>
+#include <QScriptEngine>
+
+#include "controllers/colormapper.h"
+
+class ColorMapperJSProxy final : public QObject {
+    Q_OBJECT
+  public:
+    ColorMapperJSProxy() = delete;
+    ColorMapperJSProxy(QScriptEngine* pScriptEngine, QMap<QRgb, QVariant> availableColors);
+
+    ~ColorMapperJSProxy() {
+        delete m_colorMapper;
+    };
+
+  public slots:
+    QScriptValue getNearestColor(uint ColorCode);
```

Comment:
I did, but I had to undo that commit because it will throw a runtime error.

```
Warning [Controller]: ControllerEngine: Invocation of connection  "{6642c00e-ab65-4b1c-b359-1e6426cf9a8a}" "connected to ([Channel1], hotcue_1_enabled) failed:" "TypeError: cannot call getNearestValue(): argument 1 has unknown type `QRgb' (register the type with qScriptRegisterMetaType())"
Warning [Controller]: ControllerEngine: Invocation of connection  "{fc088244-7bb5-49b6-96ce-5d4cd91268fd}" "connected to ([Channel1], hotcue_1_color) failed:" "TypeError: cannot call getNearestValue(): argument 1 has unknown type `QRgb' (register the type with qScriptRegisterMetaType())"
Warning [Controller]: ControllerEngine: "Nicht abgefangene Ausnahme bei Zeile 340 in Datei /home/jan/.mixxx/controllers/midi-components-0.0.js: TypeError: cannot call getNearestValue(): argument 1 has unknown type `QRgb' (register the type with qScriptRegisterMetaType())"
```